Sources/Purchasing/StoreKit2/StoreKit2Receipt.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2Receipt.swift
+//
+//  Created by MarkVillacampa on 26/10/23.
+
+import StoreKit
+
+/// A type that resembles the structure of a StoreKit 1 receipt using StoreKit 2 data.
+struct StoreKit2Receipt: Equatable, Codable {
+
+    /// The server environment where the receipt was generated.
+    let environment: StoreEnvironment
+
+    /// The current subscription status for each subscription group, including the renewal information.
+    /// The keys of the dictionary represent Subscription Group Identifiers, where the values are arrays 
+    /// of JWS tokens of the renewal information.
+    let subscriptionStatus: [String: [String]]
```

Comment:
Thinking about adding some typealias to String, to improve clarity, thoughts?

Reply:
A typealias doesn't provide any more guarantees. Can this by its own `struct`, and convert it to this format only when encoding it?

Reply:
+1 for structs

Reply:
also nitpick if it stays as dict: name dictionaries `valuesByKey` so it's clear what to expect from both sides, i.e.: `subscriptionStatusBySubscriptionGroupId`